### ISSUE TITLE
False positive on unused path value

### ISSUE THREAD

**Issue by tfwright** [rrrene/credo#1005]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals ✔️ 
* For bugs, please do a quick search and make sure the bug has not yet been reported here ✔️ 

### Environment

* Credo version (`mix credo -v`): 1.6.7
* Erlang/Elixir version (`elixir -v`): 1.13
* Operating system: OSX

### What were you trying to do?

Pass result of path to function with side effects

### Expected outcome

Pass credo check

### Actual outcome

Error when result of function is not assigned to var

This worked

`:ok = function_with_side_effects(Path.join(...))`

This didn't

`function_with_side_effects!(Path.join(...))`

The value is being handled the same in either case.

**Comment by rrrene** [comment]
Hi, I wasn't able to reproduce this.

I added a test, trying to reproduce this in different environments (Elixir/OTP versions).

Is this still a problem with Credo v1.6.7?

**Comment by tfwright** [comment]
Hi, I don't recall the context where I originally encountered this, I haven't encountered it since, and just failed to reproduce in an existing project on 1.6.7 so it might have been some other error on my side. Feel free to close, and sorry for the confusion.

**Comment by rrrene** [comment]
> and sorry for the confusion

Don't apologize 👏  

Better to report a bug too many than too few (this particular bug might have been fixed recently in v1.6.6).

**Comment by tfwright** [comment]
So, I did actually encounter this again, and unfortunately it does seem to still be an issue on 1.6.7.

Here is something similar to the offending code:

```
    MyApp.runtime_mod().exec!("system-util", [
      arg_a,
      Path.join(path_a, path_b)
    ])
```

**Comment by tfwright** [comment]
# Credo false positive

```elixir
Mix.install([
  {:credo, "~> 1.6.7"}
])
```



```
Resolving Hex dependencies...
Dependency resolution completed:
New:
  bunt 0.2.1
  credo 1.6.7
  file_system 0.2.10
  jason 1.4.0
* Getting credo (Hex package)
* Getting bunt (Hex package)
* Getting file_system (Hex package)
* Getting jason (Hex package)

13:08:13.866 [info] Compiling file system watcher for Mac...

13:08:14.213 [info] Done.
==> file_system
Compiling 7 files (.ex)
Generated file_system app
==> bunt
Compiling 2 files (.ex)
Generated bunt app
==> jason
Compiling 10 files (.ex)
Generated jason app
==> credo
Compiling 242 files (.ex)
Generated credo app
```



```
:ok
```

## Section

```elixir
import Credo.Test.Case

"""
defmodule CredoSampleModule do
  def my_func do
    MyApp.runtime_mod().exec!("system-util", [
      arg_a,
      Path.join(path_a, path_b)
    ])

    :ok
  end
end
"""
|> to_source_file()
|> run_check(Credo.Check.Warning.UnusedPathOperation)
```



```
[
  %Credo.Issue{
    check: Credo.Check.Warning.UnusedPathOperation,
    category: :warning,
    priority: 11,
    severity: 1,
    message: "There should be no unused return values for Path functions.",
    filename: "test-untitled-576460752303396223.ex",
    line_no: 5,
    column: 7,
    exit_status: 16,
    trigger: "Path.join",
    diff_marker: nil,
    meta: [],
    scope: "CredoSampleModule.my_func"
  }
]
```
